src/OpenApi/src/Services/OpenApiDocumentService.cs

```diff
@@ -73,18 +69,27 @@ public async Task<OpenApiDocument> GetOpenApiDocumentAsync(IServiceProvider scop
         var document = new OpenApiDocument
         {
             Info = GetOpenApiInfo(),
-            Paths = await GetOpenApiPathsAsync(capturedTags, scopedServiceProvider, operationTransformers, schemaTransformers, cancellationToken),
-            Servers = GetOpenApiServers(),
-            Tags = [.. capturedTags]
+            Servers = GetOpenApiServers()
         };
+        document.Paths = await GetOpenApiPathsAsync(document, scopedServiceProvider, operationTransformers, schemaTransformers, cancellationToken);
+        document.Tags = document.Tags?.Distinct(OpenApiTagComparer.Instance).ToList();
         try
         {
             await ApplyTransformersAsync(document, scopedServiceProvider, cancellationToken);
         }
+
         finally
         {
             await FinalizeTransformers(schemaTransformers, operationTransformers);
         }
+        // Call register components to support
+        // resolution of references in the document.
+        document.Workspace ??= new();
+        document.Workspace.RegisterComponents(document);
+        if (document.Components?.Schemas is not null)
```

Comment:
We do this to ensure that the schemas are always ordered lexicographically by reference ID in the document. This was previously done in the `OpenApiSchemaReferenceTransformer` but since that's been removed in favor of using `OpenApiSchemaReference` directly, we need to do the sorting somewhere.